src/main/java/com/parsing/service/ProzorroParserService.java

```diff
@@ -0,0 +1,178 @@
+import com.parsing.model.Status;
+import com.parsing.parsers.prozorro.URLListBuilder;
+import com.parsing.repository.LotResultRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RequiredArgsConstructor
+@Slf4j
+@Service
+public class ProzorroParserService {
+
+    private static final String NOT_PRESENT = "NOT PRESENT";
+    private static final String TENDER_NOT_FOUND = "Тендер не знайдено | ProZorro";
+    private static int MAX_COUNT = 500;
+    private static final List<List<String>> urls = URLListBuilder.buildListURLS();
+    private static final List<String> LAPTOPS_DK = List.of("30230000", "30210000");
+    private final LotResultRepository lotResultRepository;
+
+    public void parse() {
+        for (List<String> day : urls) {
+            MAX_COUNT = 500;
+            for (int x = 0; x < day.size(); x++) {
+                log.debug("Starting parsing for url " + day.get(x));
+                Document document = null;
+                int counter = 1;
+                document = getDocument(day.get(x), document, counter);
+                Element element = getElement(document);
+                if (isTenderNotFound(document) < 1) {
+                    x = day.size();
+                } else if (Objects.nonNull(element)) {
+                    LotResult lot = new LotResult();
+                    String Dk = parseDK(document);
+                    if (Dk != null && LAPTOPS_DK.contains(Dk)) {
+                        lot.setDk(parseDK(document));
+                        lot.setUrl(day.get(x));
+                        lot.setPdfLink(parsePDFLink(document));
+                        Status status = parsePDFLink(document).equals(NOT_PRESENT) ? Status.CREATED : Status.PARSED;
+                        lot.setStatus(status);
+                        lot.setPrice(parsePrice(document));
+                        lot.setParsingDate(parseDate(day.get(x)));
+                        lot.setLotPDFResult(new LotPDFResult());
+                        saveLotResult(day.get(x), lot);
+                    }
+                }
+
+            }
+
+        }
+    }
+
+    @Transactional
+    public void saveLotResult(String url, LotResult lot) {
+        lotResultRepository.saveAndFlush(lot);
+        log.debug("parsed URL = " + url);
+    }
+
+
+    private Element getElement(Document document) {
+        Element element = null;
+        try {
+            element = document.getElementsByClass("infobox-link").first();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
log warn with some information that will help us to understand what's going on.

Reply:
Done.